### ISSUE TITLE
Close doesn't work and dialog is opened 4 times

### ISSUE THREAD

**Issue by archasek** [likeastore/ngDialog#319]
Hi.
I'm using a directive, dialog is properly shown. But when I click close, nothing happens. ngDialog.opened works (console.log is triggered 4 times!), ngDialog.closing works (console.log 5 times), ngDialog.close doesn't work.
I'm using an icon in my template:
```
        <i class="icon icon--close" data-ng-click="closeThisDialog('Close')">close dialog</i>
```

directive:

```
    .directive('a', ['$rootScope', 'ngDialog', function ($rootScope, ngDialog) {
      return {
        restrict: 'E',
        link: function(scope, element, attrs) {
          var href = attrs.href.toLowerCase();

          if (href == '#/terms' || href == '#terms') {
            element.on('click', function(e) {
              if (element.parents('.main-footer').length == 0) {
                e.preventDefault();

                ngDialog.open({
                  template: 'templates/terms.html',
                  controller: 'TermsController',
                  id: 'termsDialog',
                  name: 'termsDialog',
                  showclose: false,
                  appendTo: '[data-ng-view]',
                  preCloseCallback: function(value) {
                    return true;
                  }
                }); 
              }
            });
          }
          $rootScope.$on('ngDialog.opened', function (event, $dialog) {
            console.log($dialog.dialog)
            $dialog.dialog.prevAll('.container').hide();
          });

          $rootScope.$on('ngDialog.closed', function (event, $dialog) {
            $dialog.dialog.prevAll('.container').show();
          });

          $rootScope.$on('ngDialog.closing', function (event, $dialog) {
console.log('closing');
          });
        }
      };
    }])
```

Screenshot of my console after opening and closing a dialog:

![image](https://cloud.githubusercontent.com/assets/2247211/9818913/23c7c0be-58ab-11e5-9df2-726d0762359c.png)

Can you help me?

**Comment by faceleg** [comment]
Could you please put together a bare bones example that shows this bug in either jsfiddle or plunker?

**Comment by archasek** [comment]
It's something like this: https://jsfiddle.net/tmvjf8bo/
The behaviour is quite different - locally "closing" event triggering works, on jsfiddle it doesn't. And on jsfiddle dialog is fired only once, locally 4 times O_o.

**Comment by faceleg** [comment]
Your `closeThisDialog` function doesn't exist in the fiddle

**Comment by archasek** [comment]
How to close dialog then? How can I make it work? And what is:

![image](https://cloud.githubusercontent.com/assets/2247211/9872900/943f3cde-5b9c-11e5-828e-c7210bf17374.png)

Thanks for help.

**Comment by faceleg** [comment]
Does the same thing happen if you use a template file instead of inline?

**Comment by archasek** [comment]
Yes.

**Comment by faceleg** [comment]
Sounds like a bug. When I have a chance I will try to replicate it with a unit test, then look at fixing it.

**Comment by archasek** [comment]
OK thanks, I'll be waiting.

**Comment by archasek** [comment]
I've recognized a new issue, maybe related to our main bug. The close button doesn't work even for plain ngDialog popup (without `appendTo`). Escape key and clicking outside the popup don't work as well. 

I recorded a short video to show you this error (swf): http://www.fastswf.com/QTMXbHY

This is my code (I pasted an excerpt):

```
          if (/\.(jpg|png|gif)$/.test(href)) {
            element.on('click', function(e) {
              e.preventDefault();

              ngDialog.open({
                template: '<img src="' + href + '" alt="' + element.text() + '" />',
                plain: true,
                name: 'imageDialog',
                className: 'ngdialog-theme-plain',
                showclose: true
              });
            });
          }

          $rootScope.$on('ngDialog.opened', function (event, $dialog) {
            console.log($dialog.dialog)
            console.log('opened');
            $dialog.dialog.prevAll('.container').hide();
          });

          $rootScope.$on('ngDialog.closed', function (event, $dialog) {
            $dialog.dialog.prevAll('.container').show();
            console.log('closed');
          });

          $rootScope.$on('ngDialog.closing', function (event, $dialog) {
            console.log('closing');
          });
```

Console (closing is fired 5times at start and then it's not fired even once, when I click close button):

![image](https://cloud.githubusercontent.com/assets/2247211/9905565/f50caf6c-5c85-11e5-881e-aa6cad40cad8.png)

**Comment by archasek** [comment]
Hey @faceleg, any update? Or maybe do you know when you will be able to try to fix it?

**Comment by archasek** [comment]
It doesn't work.

**Comment by faceleg** [comment]
This unit test: https://github.com/likeastore/ngDialog/blob/master/tests/unit/ngDialog_closeThisDialog.js tests that `closeThisDialog` exists on the scope. 

Strip everything not directly related to ngDialog from your test case and re-try.

**Comment by archasek** [comment]
Ehhh, sadly, I will have to use my workaround :(

**Comment by faceleg** [comment]
If you do get time to help narrow this down, let me know!

**Comment by archasek** [comment]
@faceleg I've just recognized that closing works on IE9 and doesn't work on my Chrome. Surprise.

**Comment by faceleg** [comment]
yes a big surprise! What is the error you get on chrome?

**Comment by archasek** [comment]
Nothing changed, everything I get is described here and in #321

**Comment by robsimpkins** [comment]
A little late to the party, but I experienced a similar issue. The root of my problem appeared to surround animations. I had removed them from the CSS, rather than use the disableAnimations option. This meant that the animation end event was never fired and the dialog not closed.

Restoring the CSS and/or using the disableAnimations option fixed the problem for me.

**Comment by fultonm** [comment]
disableAnimations option fixed my problem. Can we have this default to true for now?

**Comment by robsimpkins** [comment]
The disable animations issue I encountered wasn't a bug. It was user error on my part.
